### PR TITLE
Tweak Python sanity test

### PR DESCRIPTION
**Pull request by nathanielmanistaatgoogle**

### CODE REVIEW

File: src/python/grpcio_tests/tests/_sanity/_sanity_test.py [hunk 1]
```diff
@@ -21,24 +21,26 @@
...
         loader = tests.Loader()
         loader.loadTestsFromNames(['tests'])
-        test_suite_names = [
+        test_suite_names = (
             test_case_class.id().rsplit('.', 1)[0]
             for test_case_class in tests._loader.iterate_suite_cases(
                 loader.suite)
-        ]
+        )
         test_suite_names = sorted(set(test_suite_names))
```
Comment (mehrdada) [inline src/python/grpcio_tests/tests/_sanity/_sanity_test.py]: Please consider merging the two assignments into one `test_suite_names = sorted(set(...))`.

Comment (nathanielmanistaatgoogle) [reply src/python/grpcio_tests/tests/_sanity/_sanity_test.py]: Good call; done with a set comprehension.

File: src/python/grpcio_tests/tests/_sanity/_sanity_test.py [hunk 2]
```diff
@@ -21,24 +21,25 @@
 import tests
 
 
-class Sanity(unittest.TestCase):
+class SanityTest(unittest.TestCase):
+
+    maxDiff = 32768
```
Comment (mehrdada) [inline src/python/grpcio_tests/tests/_sanity/_sanity_test.py]: Wait... what is this supposed to do?

Comment (nathanielmanistaatgoogle) [reply src/python/grpcio_tests/tests/_sanity/_sanity_test.py]: I mentioned it in the commit log message, though perhaps not in as much detail as I could have for readers who haven't before heard of it. It overrides a `unittest.TestCase` attribute that controls how much to print when an `assert<Something>` method fails. In our case in this test the list of tests can be as much as five whole kilobytes, and I got tired of seeing on test failure the "we've truncated the diff that you're seeing" message. Try it out for yourself by putting a typo into `tests.json` and then running `_sanity_test`?

Comment (mehrdada) [reply src/python/grpcio_tests/tests/_sanity/_sanity_test.py]: I see. Thanks. Though being able to silently override object *fields* like that (and not as an assignment in *constru*...*em*...initializer) doesn't make me hate python any less.

Comment (nathanielmanistaatgoogle) [reply src/python/grpcio_tests/tests/_sanity/_sanity_test.py]: It's uh... it's an object model. Definitely not the one I'd first choose either.

**Pull request merged**